pybamm/solvers/base_solver.py

```diff
@@ -1160,11 +1160,34 @@ def step(
             raise pybamm.SolverError(
                 f"Step time must be at least {pybamm.TimerTime(step_start_offset)}"
             )
-
         t_start = old_solution.t[-1]
```

Comment:
originally we had 

```python3
        # Make sure dt is greater than the offset
        step_start_offset = pybamm.settings.step_start_offset
        if dt <= step_start_offset:
            raise pybamm.SolverError(
                f"Step time must be at least {pybamm.TimerTime(step_start_offset)}"
            )

        t_start = old_solution.t[-1]
        t_end = t_start + dt
        # Calculate t_eval
        t_eval = np.linspace(t_start, t_end, npts)
```

this can be replaced with

```python3
        # Make sure dt is greater than the offset
        step_start_offset = pybamm.settings.step_start_offset
        if dt <= step_start_offset:
            raise pybamm.SolverError(
                f"Step time must be at least {pybamm.TimerTime(step_start_offset)}"
            )

        if non_linear_time is not None:
            # Checking if non_linear_time lies within range
            if non_linear_time[0] != 0 or non_linear_time[-1] != dt:
                raise pybamm.SolverError(
                    "Elements inside array non_linear_time must lie in the closed interval 0 to dt"
                )
        else:
            non_linear_time = np.array([0, dt])

        t_start = old_solution.t[-1]
        t_eval = t_start + non_linear_time
```

and i think achieves what you are trying to do